### PR TITLE
Add test to validate #37567.

### PR DESCRIPTION
**Pull request by JeremyKuhne**

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
`Cookies_GetProperty_ReturnsExpected` failed on Mac with a connection refused error.

File: src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs [hunk 1]
```diff
@@ -67,6 +67,26 @@ public static void PrivateSetter()
             Assert.Null(objCopy.GetMyString());
         }
 
+        [Fact]
+        public static void PrivateSetterPublicGetter()
+        {
+            // https://github.com/dotnet/corefx/issues/37567
```
Comment (ahsonkhan) [inline src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs]: Can we also try to serialize/deserialize and verify round-tripping makes sense in this case?

File: src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs [hunk 2]
```diff
@@ -67,6 +67,26 @@ public static void PrivateSetter()
...
+            ClassWithPublicGetterAndPrivateSetter obj
+                = JsonSerializer.Parse<ClassWithPublicGetterAndPrivateSetter>(@"{ ""Class"": {} }");
```
Comment (ahsonkhan) [inline src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs]: Ideally, we should be testing all overloads for these test casess (UTF-8 bytes, async stream, etc.) and same with the serializer side, but that's outside the scope of this PR. cc @steveharter

**Comment by JeremyKuhne** [comment]
#37567

File: src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs]: I'm not sure how useful that is in the private case?

**Pull request merged**